mobify/platform-scaffold
# Add standalone flag to redux store

This PR adds a flag to the redux store that indicates if the user is viewing the web app in standalone/homescreen mode.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1280
 **Linked PRs**: (links to corresponding PRs, optional)

## Changes
- Added util to detect if the PWA is in standalone mode
- Added an action that will set `standaloneApp` flag 

## Todos
- [ ] Remove test console.log

## How to test-drive this PR
- in Charles enable SSL Proxying (Proxy menu -> SSL Proxy Settings)
![screen shot 2017-08-14 at 11 58 25 am](https://user-images.githubusercontent.com/1301857/29287071-0ee716d2-80e9-11e7-86d7-ed9151efff06.png)
- still in Charles, set up the following rule for map remote (Tools Menu -> Map Remote) 
![screen shot 2017-08-17 at 1 25 32 pm](https://user-images.githubusercontent.com/1301857/29432261-c7aea7a8-834f-11e7-9e3c-0524a42ce5de.png)
- Run `npm start`
- get an android device (ex. Pixel)
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1) in your browser
- open the browser menu and click add to home screen
- go to the homescreen and open the PWA from the homescreen button
- Inspect the standalone instance
- you should see the following in the console: "****** Standalone flag set to: true"
